### PR TITLE
Minor dedupe xla/pxla tracing code.

### PR DESCRIPTION
**Pull request by chr1sj0nes**

### CODE REVIEW

File: jax/interpreters/pxla.py [hunk 1]
```diff
@@ -460,12 +460,10 @@ def dynamic_fun(dummy, *args):
   pvals = [pe.PartialVal((aval, core.unit)) for aval in avals]
   # We add a dummy first invar, to carry the trace details to `dynamic_fun`
   pval = pe.PartialVal([core.abstract_unit, core.unit])  # dummy value for axis env
-  with core.new_master(pe.StagingJaxprTrace, True) as master:
-    jaxpr, (out_pvals, consts, env) = pe.trace_to_subjaxpr(
-        dynamic_fun, master, False).call_wrapped([pval] + pvals)
-    jaxpr.invars = jaxpr.invars[1:]  # ignore dummy
-    assert not env
-    del master
+  jaxpr, out_pvals, consts = pe.trace_to_jaxpr(
```
Comment (gnecula) [inline jax/interpreters/pxla.py]: maybe it is better to pass `instantiate=False` here explicitly, for clarity?

**Review by gnecula — CHANGES_REQUESTED** [review]
I think this is a good change. There is too much code duplication in the tracing of the various call primitives. 

I would recommend applying the same treatment to sharded_jit (that code is WIP, but it would benefit from this as well)

File: jax/interpreters/pxla.py (hunk 1 above)
Comment (chr1sj0nes) [reply jax/interpreters/pxla.py]: Done

**Comment by gnecula** [comment]
Hey, are you planning to add this support in sharded_jit also?

**Comment by chr1sj0nes** [comment]
> Hey, are you planning to add this support in sharded_jit also?

How would I do that? Should I put in a pull request onto Skye's branch? Do you want me to do that before this gets merged?

**Comment by gnecula** [comment]
The sharded_jit is already in the interpreters directory. Let's ask @skye if she thinks there is value in adding this improvement there, or she will add it herself in the final version.

**Comment by chr1sj0nes** [comment]
> The sharded_jit is already in the interpreters directory. Let's ask @skye if she thinks there is value in adding this improvement there, or she will add it herself in the final version.

Ah. I didn't realise that was already in. I've added the change there too.

**Review by gnecula — APPROVED** [review]

**Pull request merged**